### ISSUE TITLE
[Bug]: wrong gains for weight initialization

### ISSUE THREAD

**Issue by OliEfr** [DLR-RM/stable-baselines3#1559]
### 🐛 Bug

The recommended gains for the weight init depend on the used activation function, see [torch docs](https://pytorch.org/docs/stable/nn.init.html). However, as for now the used gains are statically implemented and always the same in ActorCriticPolicies. See [here](https://github.com/DLR-RM/stable-baselines3/blob/ffe26ccf95d7e3b37067bd81025d3b4b45825038/stable_baselines3/common/policies.py#L589).

I recommend making the gains dependent on the activation function used as recommended by torch.

If you agree that this would be a useful feature, I would implement it myself and PR.

Thanks and a good day!

### To Reproduce

--

### Relevant log output / Error message

```shell
--
```

### System Info

--

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
those gains are for orthogonal initialization only (https://pytorch.org/docs/stable/_modules/torch/nn/init.html#orthogonal_), when they are not used, the default pytorch initialization is used.

The gains are from OpenAI Baselines, to keep results consistent, but compared to other initialization, I didn't see any investigation on the effect of the gain so far (this would be already a good contribution), or at least if using tanh/relu with constant gain has an effect.

**Comment by OliEfr** [comment]
Yes, I am talking about orthogonal init. I agree that it is useful to keep it consistent with OpenAI Baselines. A study regarding the effect of gain towards convergence will be useful.

It seems a coincidence (?) that the standard gain listed for ReLU for any initialization is also `sqrt(2)` [Link](https://pytorch.org/docs/stable/nn.init.html). (The gain implemented in OpenAI Baselines and sb3 is also `sqrt(2)`. Maybe they just used ReLU by default and never investigated the gain?)

One study that partly investigates impact of weight init is [this](https://arxiv.org/pdf/2006.05990v1.pdf). They find:

> initializing the policy MLP with smaller weights in the last layer

>  network initialization scheme (C56) does not matter too much
